### PR TITLE
Lint with awesome-lint via GitHub Actions

### LINKED ISSUE #1619: Lint Awesome itself
https://github.com/sindresorhus/awesome/pull/1394 went stale. Would be nice to finish it at some point. However, [awesome-lint](https://github.com/sindresorhus/awesome-lint) could probably be made better first.

### PR DESCRIPTION
**Pull request by Richienb**
See https://github.com/sindresorhus/awesome/pull/1696

Fixes #1619 
Fixes #1675 

// @sindresorhus

### CODE REVIEW

**Comment by sindresorhus** [comment]
Any comments on https://github.com/sindresorhus/awesome/pull/1696#issuecomment-585335322 ?

**Comment by ibnesayeed** [comment]
> Any comments on [#1696 (comment)](https://github.com/sindresorhus/awesome/pull/1696#issuecomment-585335322) ?

I do not see a reason why it will not pass. What issues do you anticipate will happen when merged? As far as I know, `actions/checkout@v2` had a breaking change, but that was taken care by setting the `fetch-depth` to `0`.

**Comment by Scrum** [comment]
When trying to install `awesome-lint` globally i faced the problem:
```bash
checkPermissions Missing write access to /usr/local/lib/node_modules
```
I managed to solve this by setting a custom global scope. This is how my solution looks like:

```yml
name: Awesome readme lint
on:
  pull_request:
    types: [opened, synchronize]
    branches:
      - main
jobs:
  build:
    name: awesome readme lint
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v2
      - run: |
          mkdir ~/.npm-global
          npm config set prefix '~/.npm-global'
          PATH=~/.npm-global/bin:$PATH
          npm install --global awesome-lint
          awesome-lint
```

**Comment by probonopd** [comment]
I was able to run the linter by just adding the one file from this pull request to my repository at https://github.com/AppImage/awesome-appimage/commit/1494277a41c329e2f425595c5d512880dde3f3ed without any further changes. The linter seems to work since it has found some issues.

So to me, this appears to get the job done! :+1:

**Comment by RayBB** [comment]
@probonopd your solution looks good!

You could make it a bit shorter by using `npx` like [this](https://github.com/RayBB/awesome-social-enterprise/blob/main/.github/workflows/main.yml#L27) if you want to save one step.

**Comment by KeyboardInterrupt** [comment]
Having this makes total sense, I set up `awesome-lint` on [awesome-ansible](https://github.com/KeyboardInterrupt/awesome-ansible/blob/main/.github/workflows/awesome-lint.yml) as well, although slightly different.
The main difference is that I use `npx` directly, [as described here in the `awesome-lint` projects readme](https://github.com/sindresorhus/awesome-lint#github-actions).

:+1:

**Comment by sindresorhus** [comment]
> I do not see a reason why it will not pass. What issues do you anticipate will happen when merged? As far as I know, actions/checkout@v2 had a breaking change, but that was taken care by setting the fetch-depth to 0.

I think I explained it pretty clearly in the linked comment. The readme here does not pass all the awesome-lint rules.

**Comment by sindresorhus** [comment]
@Richienb Are you interested in finishing this or should I close?

**Comment by Richienb** [comment]
I'm just coming back to this now and I'll continue working on this.

**Comment by fityanos** [comment]
missing the unicorn text @Richienb

**Comment by Richienb** [comment]
@fityanos That guideline is under the section labelled [Requirements for your Awesome list](https://github.com/sindresorhus/awesome/blob/0c86e0d3206baff23221baf006b541870dbc3151/pull_request_template.md#requirements-for-your-awesome-list). This PR does not add an Awesome list.

**Comment by fityanos** [comment]
@Richienb I'm highlighting this
image

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - uses: actions/setup-node@v1
+        with:
+          node-version: 12
+      - run: npm install --global awesome-lint
+      - run: awesome-lint
```
Comment (zrosenbauer) [inline .github/workflows/main.yml:15]: Could just use `npx awesome-lint` per the docs, seem redundant having multiple steps.

**Comment by zrosenbauer** [comment]
@Richienb you trying to finish this up? I'd be happy to jump in and help here. I'm assuming @sindresorhus we could use the remark comments to filter out some of the issues with the list? 

Screen Shot 2021-02-10 at 7 32 42 PM

I was able to add `<!--lint ignore awesome-github awesome-heading-->` to nix majority of those. Need to fix code of conduct still...

**Comment by Richienb** [comment]
I'll close this PR if #1851 gets merged

**Review by Danix421 — APPROVED** [review]
I don't have a Awersome list but if i'm correct that is good

**Comment by pgollucci** [comment]
I am about to submit the awesome-projen list based on projen.  I am also the creator of p6-projen-project-awesome-list which uses projen to setup a repo as  an awesome-list and "obsoletes" yeoman with a more general purpose tool.

It already includes awesome-lint in the the GitHub Action workflow.  As far as I know, it also sets up the repo in 100% compliance out of the box.

**Comment by pgollucci** [comment]
https://github.com/p6m7g8/awesome-projen
https://github.com/p6m7g8/p6-projen-project-awesome-list
https://github.com/projen/projen

projen is related to awesome-cdk and awesome-cdk8s and awesome-terraform-cdk

**Pull request closed without merging**